### PR TITLE
fix: add k8s.io/client-go/.../azure missing module

### PR DESCRIPTION
**Pull request by sylr**
As this module is not called directly it was omitted but it is needed by the upgrade process when using AAD auth.

```
INFO[0402] Error validating upgraded master VM: k8s10-euw-sandbox-master-0
FATA[0402] Error upgrading cluster: No Auth Provider found for name "azure"
```

### CODE REVIEW

**Comment by tariq1890** [comment]
Is this because you are dependent on the method below @sylr ?

```
func init() {
	if err := restclient.RegisterAuthProviderPlugin("azure", newAzureAuthProvider); err != nil {
		klog.Fatalf("Failed to register azure auth plugin: %v", err)
	}
}
```

**Comment by tariq1890** [comment]
Not sure how I feel about this. Forcing an import because of an init-method is a huge antipattern for Go.

File: Gopkg.toml [hunk 1]
```diff
@@ -96,15 +96,15 @@ required = [
 
 [[constraint]]
   name = "k8s.io/api"
-  version = "kubernetes-1.10.0"
+  version = "kubernetes-1.13.2"
```
Comment (tariq1890) [inline Gopkg.toml]: We have currently put the k8s version updates on hold. @jackfrancis can provide details about this.

File: pkg/armhelpers/kubeclient.go [hunk 2]
```diff
@@ -14,6 +14,7 @@ import (
 	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/azure"
```
Comment (tariq1890) [inline pkg/armhelpers/kubeclient.go]: :(

**Comment by jackfrancis** [comment]
I'm a little confused why the AAD upgrade errors require a rev of client-go

**Comment by sylr** [comment]
As far as I remember it was not compiling anymore with the current pinned versions when I forced the import of the module. 

Updating everything to 1.13.2 fixed it.

**Comment by sylr** [comment]
> Is this because you are dependent on the method below @sylr ?
> 
> ```
> func init() {
> 	if err := restclient.RegisterAuthProviderPlugin("azure", newAzureAuthProvider); err != nil {
> 		klog.Fatalf("Failed to register azure auth plugin: %v", err)
> 	}
> }
> ```

Yes

**Comment by sylr** [comment]
> Not sure how I feel about this. Forcing an import because of an init-method is a huge antipattern for Go.

Is it ? I'm new to Go but I've seen this quite a few times in several projects.

Anyway do you know any other way to fix that ?

**Comment by sylr** [comment]
I remember now, I upgraded the libs because I needed https://github.com/kubernetes/client-go/commit/af076e5342ee3b17ac7f68b33f1b48ab42157774#diff-ccba71f00b34f4ba0c9297d2f8e5d808 to make it compile.

It is only included since kubernetes-1.13.0-beta.0.

**Comment by tariq1890** [comment]
> Anyway do you know any other way to fix that ?
I saw the client-go code, looks like that pattern is baked in :(.

> I remember now, I upgraded the libs because 
Did you use a patched vendor of client-go in your aks-engine branch ?

**Comment by sylr** [comment]
> Did you use a patched vendor of client-go in your aks-engine branch ?

A patched vendor ? I just updated the Gopkg.toml and ran `dep ensure`

**Pull request closed without merging**